evals/elsuite/basic/fuzzy_match.py

```diff
@@ -19,11 +21,16 @@ def __init__(
 
     def eval_sample(self, test_sample, rng):
         prompt, correct_answers = test_sample["input"], test_sample["ideal"]
+        plugins = Plugin.load(test_sample.get("plugins", []))
+        plugin_actions = PluginAction.load(test_sample.get("plugin_actions", []))
```

Comment:
A few reasons, that I hold weakly:
- Testing 'meta' properties of tool use, such as "This evaluates the ability of the model to use the various search engines in the world".  Sort of like going through a toolkit and asking the model "Can you use the hammer", "Can you use the jigsaw", "Can you use the stapler" to assess how well it can use the entire kit
- Testing performance of an eval against multiple plugins that do approximately the same thing.  An example would be a search evaluation where we test all combinations of the major search engines.

I think there's a fair argument to be made that those should be separate evaluations, though, combined into a set if we think a grouping is meaningful.  i.e. an information factuality eval set with [No Search Engine, Search Engine A, Search Engine B, Search Engine A+B] as the four evaluations it is composed of. 

Thinking it through a bit more, most of my concerns are well-handled via rolling the evaluation up into a set, so let me move the `plugin` initialization into the `__init__` function for now

Reply:
Hmm, we still need to init one set of plugins per sample (So that we keep the state separated), but `__init__` doesn't know how many samples it is processing.  Given that, initializing in `eval_sample` seems like the most natural location. 

It isn't exactly clear to me if we want to set the enabled plugins for the eval at the `sample` or `eval` level, though